sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -418,10 +415,27 @@ public async virtual Task InvokeFailedArg(Exception ex)
                         ClientDiagnostics)
                         .ConfigureAwait(false);
                 }
+                _progressTracker.IncrementFailedFiles();
+
+                // Update the JobPartStatus. If was already updated (e.g. there was a failed item before)
+                // then don't raise the PartTransferStatusEventHandler
```

Comment:
This event is intended to be raised once and only once for a transfer, regardless of number of failed items? Is there a different event that happens for every failed item?

Reply:
Yup, we only want to raise this event once, regardless of number of failed items.

If they want it for every time we see a Failed Item, the user should subscribe to the `DataTransferOptions.ItemTransferFailed` event handler.